python/lsst/ip/isr/isrTaskLSST.py

```diff
@@ -1238,10 +1330,28 @@ def run(self, ccdExposure, *, dnlLUT=None, bias=None, deferredChargeCalib=None,
 
         gains = ptc.gain
 
+        # Validation step: check inputs match exposure configuration.
+        exposureMetadata = ccdExposure.getMetadata()
+        self.compareCameraKeywords(exposureMetadata, ptc, "PTC")
+        if self.doLinearize(detector):
+            self.compareCameraKeywords(exposureMetadata, linearizer, "linearizer")
+        if self.config.doBias:
+            self.compareCameraKeywords(exposureMetadata, bias, "bias")
+        if self.config.doCrosstalk:
+            self.compareCameraKeywords(exposureMetadata, crosstalk, "crosstalk")
+        if self.config.doDeferredCharge:
+            self.compareCameraKeywords(exposureMetadata, deferredChargeCalib, "CTI")
+        if self.config.doDefect:
+            self.compareCameraKeywords(exposureMetadata, defects, "defects")
+        if self.config.doDark:
+            self.compareCameraKeywords(exposureMetadata, dark, "dark")
+        if self.config.doBrighterFatter:
+            self.compareCameraKeywords(exposureMetadata, bfKernel, "brighter-fatter")
+        if self.config.doFlat:
+            self.compareCameraKeywords(exposureMetadata, flat, "flat")
+
         if self.config.doHeaderProvenance:
-            # Inputs have been validated, so we can add their date
-            # information to the output header.
-            exposureMetadata = ccdExposure.getMetadata()
+            # Add calibration date information to the output header.
```

Comment:
We'll need to add the other `doHeaderProvenance` code, from `isrTask.py:L1146`.  I think I had assumed this could be added to the `run` method, but looking at that code now, I think it needs to be in the `runQuantum` method for this task as well, as it gets some of the information that gets inserted into the header from the butler reference.  I think that will replace the `extractCalibDate` calls here.

Reply:
I don't think you need to worry about the `additionalInputs`, though.  Since we don't mangle any of the inputs in the new `runQuantum`, everything should hopefully just work without any special cases.